### PR TITLE
Fix openning message with expired pass phrase

### LINKED ISSUE #416: app keeps forgetting all keys
reproduce:
1) set up account, choose pass phrase in memory
2) use the app for a bit
3) use it 4 hours later or overnight - open a message

expected: keys are permanently stored and remembered, the app remains usable but asks for pass phrase

actual:

image

65C387EE-FB4A-4B02-A303-C5B53A98CAC3

### LINKED ISSUE #426: decryptedKey response from Core should not be nullable
```swift
    struct DecryptKey: Decodable {
        let decryptedKey: String?
    }
```

should be

```swift
    struct DecryptKey: Decodable {
        let decryptedKey: String
    }
```

### PR DESCRIPTION
**Pull request by Kharchevskyi**
This PR fixes bug when pass phrase expired
For testing purposes timeout can be changed in `InMemoryPassPhraseStorage`

close #416 
----------------------------------
 
**Tests** :
- Difficult to test. Time consuming
--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Review by tomholub — COMMENTED** [review]
Does this still allow me to view list of keys in Settings -> Keys section when the pass phrase expired?

**Comment by Kharchevskyi** [comment]
> Does this still allow me to view list of keys in Settings -> Keys section when the pass phrase expired?

hm...good catch, I assume - no. Will fix

**Comment by tomholub** [comment]
The design of this method needs to change. It's good that it collects pass phrases, but it should not be required. It's ok to return keys missing pass phrases if they are missing. Let the downstream code deal with that.

**Review by Kharchevskyi — COMMENTED** [review]
LGTM

File: FlowCrypt/Functionality/Services/Key Services/KeyService.swift [hunk 1]
```diff
@@ -34,26 +34,19 @@ final class KeyService: KeyServiceType {
     }
 
     func retrieveKeyDetails() -> Result<[KeyDetails], KeyServiceError> {
```
Comment (tomholub) [inline FlowCrypt/Functionality/Services/Key Services/KeyService.swift:36]: The difference between `retrieveKeyDetails` and `getPrivateKeys` is not obvious from the name. Let's rename it:
 - `retrieveKeyDetails` -> `getPrvKeyDetails`
 - `getPrivateKeys` -> `getPrvKeyInfo`

Still not so obvious, but at least it highlights the different formats.

Then, please add a comment to highlight what is each method for, why do we have two similar methods.

**Review by tomholub — CHANGES_REQUESTED** [review]

**Review by tomholub — APPROVED** [review]

**Pull request merged**